### PR TITLE
search: add subcommands for tarantool/tarantool-ee/tt binaries

### LINKED ISSUE #444: Make search command behavior consistent with install/uninstall
`tt install/uninstall` use sub-commands to specify a program to work with. `tt search` should behave the same way: use sub-commands tt/tarantool/tarantool-ee with sub-command specific options.

### PR DESCRIPTION
**Pull request by vr009**
Closes #444

### CODE REVIEW

File: cli/cmd/search.go [hunk 1]
```diff
@@ -35,45 +81,33 @@ func NewSearchCmd() *cobra.Command {
...
 	searchCmd.Flags().BoolVarP(&local, "local-repo", "", false,
 		"search in local files")
-	searchCmd.Flags().BoolVar(&debug, "debug", debug,
-		"search for debug builds of tarantool-ee SDK")
-	searchCmd.Flags().StringVar(&searchCtx.ReleaseVersion, "version", searchCtx.ReleaseVersion,
-		"specify version")
+
+	searchCmd.AddCommand(
+		newSearchTarantoolCmd(),
+		newSearchTarantoolEeCmd(),
```
Comment (psergee) [inline cli/cmd/search.go:90]: So, we should remove manual options check in search implementation. For example, there is no need to check that dbg search is applicable only for ee. This is handled by sub-commands now.

File: cli/cmd/search.go [hunk 2]
```diff
@@ -35,45 +81,33 @@ func NewSearchCmd() *cobra.Command {
...
-		ValidArgs: []string{"tt", "tarantool", "tarantool-ee"},
 	}
 	searchCmd.Flags().BoolVarP(&local, "local-repo", "", false,
 		"search in local files")
-	searchCmd.Flags().BoolVar(&debug, "debug", debug,
-		"search for debug builds of tarantool-ee SDK")
-	searchCmd.Flags().StringVar(&searchCtx.ReleaseVersion, "version", searchCtx.ReleaseVersion,
-		"specify version")
+
+	searchCmd.AddCommand(
```
Comment (psergee) [inline cli/cmd/search.go:88]: Please update the changelog. These are user visible changes.

File: cli/cmd/search.go [hunk 3]
```diff
@@ -17,16 +16,63 @@ var (
...
+			err := modules.RunCmd(&cmdCtx, cmd.CommandPath(), &modulesInfo,
+				internalSearchModule, args)
+			handleCmdErr(cmd, err)
+		},
+	}
+
+	return tntCmd
+}
+
+// newInstallTarantoolEeCmd creates a command to install tarantool-ee.
```
[the remark this answers is not in the document]
Comment (vr009) [reply cli/cmd/search.go]: Fixed.

File: cli/cmd/search.go (hunk 2 above)
Comment (vr009) [reply cli/cmd/search.go:87]: Added.

File: cli/cmd/search.go (hunk 1 above)
Comment (vr009) [reply cli/cmd/search.go:89]: Removed.

**Review by psergee — APPROVED** [review]
lgtm

File: cli/search/search.go [hunk 4]
```diff
@@ -180,20 +182,10 @@ func SearchVersions(cmdCtx *cmdcontext.CmdCtx, searchCtx SearchCtx,
...
-			log.Warnf("--debug and --version options can only be used to" +
-				" search for tarantool-ee packages.")
-		}
 	} else if program == ProgramTt {
 		repo = GitRepoTT
-		if searchCtx.Filter == SearchDebug || (len(searchCtx.ReleaseVersion) > 0) {
-			log.Warnf("--debug and --version options can only be used to" +
-				" search for tarantool-ee packages.")
-		}
 	} else if program == ProgramEe {
```
[the remark this answers is not in the document]
Comment (vr009) [reply cli/search/search.go]: Removed.

**Pull request merged**